_rules/printable-characters-shortcut-ffbc54.md

```diff
@@ -0,0 +1,547 @@
+---
+id: ffbc54
+name: No keyboard shortcuts use only printable characters
+rule_type: atomic
+description: |
+  This rule checks that if keyboard shortcuts are implemented using only printable characters, then there is a mechanism to disable the shortcut, or to remap the shortcut to use one or more non-printable characters keys, or the shortcut for a user interface component is only available when that component has focus.
+accessibility_requirements:
+  wcag21:2.1.4: # Character Key Shortcuts (A)
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: further testing needed
+acknowledgements:
+  authors:
+    - João Vicente
+    - Carlos Duarte
+---
+
+## Applicability
+
+The rule applies to any [keyboard event][]: 
+ - where the attribute `key` is a [printable character][] key; and
+ - the method `getModifierState` returns `false`; and
+ - is [dispatched][] to an [event target][] within a [HTML document][].
+
```

Comment:
There should be a condition that the event is not already blocked.
Well, technically, if it's already blocked, then (**focus**) is true and the event passes the rule, but maybe we'd like it to be inapplicable instead.

Reply:
Could you elaborate on this? I'm afraid I don't understand where you're getting at :-(

Reply:
If an element listens to keyboard events and does stuff when 'a' is pressed. The element still receive the single-key event 'b'. But that event is already blocked (nothing happens).
Technically, that doesn't pass the "disable" condition: there is no instrument to disable the 'b' events. It does pass the "focus" condition: if the element isn't focused, then the 'b' events are blocked.

But maybe we want that to be in the applicability instead and make the 'b' events inapplicable instead.

Reply:
Got it! Thanks for the explanation. I think that makes sense, but I think I would prefer to make inapplicable those events that cause no "changes in content" instead of those that are "blocked". It is the same, but I find it more understandable. What do you think?